common/viperutil/config_util.go

```diff
@@ -315,7 +543,7 @@ func bccspHook(f reflect.Type, t reflect.Type, data interface{}) (interface{}, e
 // EnhancedExactUnmarshal is intended to unmarshal a config file into a structure
 // producing error when extraneous variables are introduced and supporting
 // the time.Duration type
-func EnhancedExactUnmarshal(v *viper.Viper, output interface{}) error {
+func EnhancedExactUnmarshal(c *ConfigParser, output interface{}) error {
```

Comment:
This is no longer a function around viper so it should be converted into a method on `ConfigParser`.